aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -389,10 +389,34 @@ impl<'a> CanonicalRequest<'a> {
         match canonical_headers.get(&HOST) {
             Some(header) => header.clone(),
             None => {
+                let port = uri.port();
+                let scheme = uri.scheme();
                 let authority = uri
                     .authority()
-                    .expect("request uri authority must be set for signing");
-                let header = HeaderValue::try_from(authority.as_str())
+                    .expect("request uri authority must be set for signing")
+                    .as_str();
+                let host = uri
+                    .host()
+                    .expect("request uri host must be set for signing");
+
+                // Check if port is default (80 for HTTP, 443 for HTTPS) and if so exclude it from the
+                // Host header when signing since RFC 2616 indicates that the default port should not be
+                // sent in the Host header (and Hyper strips default ports if they are present)
+                // https://datatracker.ietf.org/doc/html/rfc2616#section-14.23
+                // https://github.com/awslabs/aws-sdk-rust/issues/1244
+                let header_value = if let (Some(scheme), Some(port)) = (scheme, port) {
```

Comment:
Maybe consider abstracting this to a function (would maybe make it more clear without the comment or at a glance)

```rs
let header_value = if is_port_scheme_default(scheme, port) {
    host
} else {
    authority
};

#[inline]
fn is_port_scheme_default(scheme, port) -> bool { ... }
```